api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1267,6 +1297,10 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
                     )
                 })
 
+            self._prepare_cluster_pillar(
```

Comment:
please do not (consider any target nodes changes here, that stage of the logic is only about preparation (e.g. prepare pillar of salt-ssh and not on target nodes). In current version even in case when setup (jump) server is one of the target nodes it will not affect another node. For remote case it would be worse - it would try to put the pillar into `/var/lib` of that not related system.

 I would suggest you to go another way:
- let core provisioner part be installed on nodes
- do something like `file.managed` with `makedirs=True` copying one of the samples (single or dual) into calculated pillar locations, as you do inside that fun:
```
        pillar_dir = config.PRVSNR_DATA_SHARED_DIR / "srv/pillar/groups/all"
        cluster_sls_path = add_pillar_merge_prefix(
            pillar_dir / 'cluster.sls'
        )
```

In other words you may consider to create a state file like [here](https://github.com/Seagate/cortx-prvsnr/blob/6da5c201a94676cc88d0008df142c78d49eab91d/api/python/provisioner/srv/salt/provisioner/configure_salt_master.sls#L22-L27) where:
- `name` might be passed via pillar (either online or may be put inside pillar file profile as for release in `_prepare_release_pillar` function).
- `source` might be put under jinja depending on nodes number